starlette/testclient.py

```diff
@@ -68,6 +68,18 @@ def __init__(self, session: "WebSocketTestSession") -> None:
         self.session = session
 
 
+class DenialResponse(Exception):
+    def __init__(
+        self,
+        status_code: int,
+        headers: typing.List[typing.Tuple[bytes, bytes]] = [],
```

Comment:
Seems clumsy -- Why not `Dict[str, str]` (or `starlette.datastructures.Headers`)?

In fact, this should probably contain a `httpx.Response` -- this would allow reusing code more easily